restricted_site_access.php

```diff
@@ -206,18 +206,52 @@ protected static function is_restricted() {
 			$blog_public = get_site_option( 'blog_public', 2 );
 		}
 
-		$user_check = is_user_logged_in();
-
-		if ( is_multisite() ) {
-			$user_id    = get_current_user_id();
-			$user_check = is_user_logged_in() && ( is_super_admin( $user_id ) || is_user_member_of_blog( $user_id ) );
-		}
+		$user_check = self::user_can_access();
 
 		$checks = is_admin() || $user_check || 2 !== (int) $blog_public || ( defined( 'WP_INSTALLING' ) && isset( $_GET['key'] ) );
 
 		return ! $checks;
 	}
 
+	/**
+	 * Check if current user has access.
+	 *
+	 * Can be short-circuited using the `restricted_site_access_user_can_access` filter
+	 * to return a value other than null (boolean recommended).
+	 *
+	 * @return bool Whether the user has access
+	 */
+	protected static function user_can_access() {
+		/**
+		 * Filters whether the user can access before any other checks as a short-circuit.
+		 *
+		 * @param bool $access Whether the user can access
```

Comment:
Maybe a full sentence for these lines (and end with a period)? `@param bool $access Whether the user can access the site.`